zkevm-circuits/src/evm_circuit/execution/push.rs

```diff
@@ -124,23 +164,50 @@ impl<F: Field> ExecutionGadget<F> for PushGadget<F> {
         offset: usize,
         block: &Block<F>,
         _: &Transaction,
-        _: &Call,
+        call: &Call,
         step: &ExecStep,
     ) -> Result<(), Error> {
         self.same_context.assign_exec_step(region, offset, step)?;
 
         let opcode = step.opcode().unwrap();
+        let num_pushed = opcode.postfix().expect("opcode with postfix");
+        let npushed = num_pushed as usize;
+
+        let bytecode = block
+            .bytecodes
+            .get_from_h256(&call.code_hash)
+            .expect("could not find current environment's bytecode");
+
+        let code_size = bytecode.codesize() as u64;
+        self.code_size
+            .assign(region, offset, Value::known(F::from(code_size)))?;
+
+        let code_size_left = (code_size - step.pc - 1) as usize;
+        self.is_out_of_range.assign(
+            region,
+            offset,
+            F::from(code_size_left as u64),
+            F::from(num_pushed as u64),
+        )?;
 
         let value = block.get_rws(step, 0).stack_value();
         self.value.assign_u256(region, offset, value)?;
 
-        let num_additional_pushed = opcode.postfix().expect("opcode with postfix") - 1;
-        for (idx, selector) in self.selectors.iter().enumerate() {
-            selector.assign(
-                region,
-                offset,
-                Value::known(F::from((idx < num_additional_pushed as usize) as u64)),
-            )?;
+        let is_out_of_bound = code_size_left < npushed;
+        let mut out_of_bound_padding = 0;
+        if is_out_of_bound {
+            out_of_bound_padding += npushed - code_size_left
+        };
```

Comment:
```suggestion
        let is_out_of_bound = code_size_left < npushed;
        let out_of_bound_padding = if is_out_of_bound {
            out_of_bound_padding = npushed - code_size_left
        } else {
            0
        };
```
remove mut only used by underling if stmt